src/core/symbology-ng/qgsfillsymbollayerv2.cpp

```diff
@@ -3430,6 +3432,11 @@ void QgsCentroidFillSymbolLayerV2::renderPolygon( const QPolygonF& points, QList
 {
   Q_UNUSED( rings );
 
+  if ( context.renderContext().expressionContext().hasVariable( "geometry_part_num" ) )
```

Comment:
What would variable( "..." ).toInt() return if the variable is not present? 0? If so yeah we can safely remove the check.